src/utils/get-resume-state.ts

```diff
@@ -15,7 +15,7 @@ export enum StorageKey {
     SAKSNUMMER = 'nav.klage.saksnummer'
 }
 
-export function getResumeState(queryParams: string, storage: Storage): ResumeState {
+export function getResumeState(queryParams: string, storage: Storage, pathName = ''): ResumeState {
```

Comment:
Jeg kommenterer her siden den andre tråden ble outdated.

En default verdi er ikke nødvendig. Vi sjekker alltid om denne er akkurat `"/oppsummering"`. Om den er `null`, `undefined`, et tall eller hva som helst annet vil det fungere helt likt.

Dessuten er `props.location.pathname` alltid en `string` fra før, så denne logikken vil aldri endre noe, men vil alltid kjøre.

Reply:
Jeg skal gå straks, men vi kan gjerne ta en prat om det i morgen, jeg tenker det er interessant med ulik tilnærming. Tanken min med å legge inn pathName som en optional her var å unngå å endre grensesnittet på funksjonen, og at den skal kunne kjøre på samme måte som tidligere også med denne endringen. Jeg tenker at å bruke egenskaper ved `props.location.pathname` blir å koble sammen `getResumeState`-funksjonen med logikken i det som kaller den, og at det er en uheldig kobling som det er bedre å unngå.

Nå er det jo riktignok tilfellet at funksjonen bare brukes et sted i koden, så sånn sett kunne jeg unngå optional og legge til pathname i alle testene, men denne måten er i alle fall min intuisjon. Men er med på at det kan endres, altså.

Reply:
Ja, jeg skjønte hva du hadde tenkt og er med på logikken der :)

Vi binder vel ikke `getResumeState` direkte med `props.location.pathname`, det er bare et krav om at den som kaller sørger for å garantere at det er en `string`. Hadde vi krevd `RouteComponentProps` hadde det vært en tettere kobling.

Hadde det vært et annet språk enn JavaScript hadde nok min intuisjon vært akkurat den samme.

Det som påvirket min vurdering er at default verdier ikke er uten kost i JS,  for vår kodebase blir dette transpilert til noe slikt som dette eksempelet:
```js
function multiply(a, b) {
  b = (typeof b !== 'undefined') ?  b : 1
  return a * b
}
```

Men nå er vi også på detaljer som ikke har så stor betydning.
Jeg approver uansett hvilket alternativ du velger :)

I morgen er jeg forresten i onboarding hele dagen...

Reply:
Ah, men det er jo et veldig godt poeng. Lærte noe nytt, takk for forklaringen! Fjerner default-verdi.